azure/services/fleetsmembers/spec_test.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleetsmembers
+
+import (
+	"context"
+	"testing"
+
+	asocontainerservicev1 "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230315preview"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+)
+
+var (
+	fakeAzureFleetsMember = asocontainerservicev1.FleetsMember{
+		Spec: asocontainerservicev1.Fleets_Member_Spec{
+			AzureName: fakeAzureFleetsMemberSpec1.Name,
+			Owner: &genruntime.KnownResourceReference{
+				ARMID: azure.FleetID(fakeAzureFleetsMemberSpec1.SubscriptionID, fakeAzureFleetsMemberSpec1.ManagerResourceGroup, fakeAzureFleetsMemberSpec1.ManagerName),
+			},
+			Group: ptr.To(fakeAzureFleetsMemberSpec1.Group),
+			ClusterResourceReference: &genruntime.ResourceReference{
+				ARMID: azure.ManagedClusterID(fakeAzureFleetsMemberSpec1.SubscriptionID, fakeAzureFleetsMemberSpec1.ClusterResourceGroup, fakeAzureFleetsMemberSpec1.ClusterName),
+			},
+		},
+	}
+	fakeAzureFleetsMemberSpec1 = AzureFleetsMemberSpec{
```

Comment:
Is there more than one fleet member spec? If not, we can probably drop the `1` at the end.